app/scripts/controllers/metametrics-data-deletion/services/services.ts

```diff
@@ -0,0 +1,45 @@
+import getFetchWithTimeout from '../../../../../shared/modules/fetch-with-timeout';
+import { DeleteRegulationAPIResponse } from '../metametrics-data-deletion';
+
+const segmentSourceId = process.env.SEGMENT_DELETE_API_SOURCE_ID;
+const segmentRegulationEndpoint = process.env.SEGMENT_REGULATIONS_ENDPOINT;
+
+const fetchWithTimeout = getFetchWithTimeout();
+
+export async function createDataDeletionRegulationTask(
```

Comment:
I would recommend converting these two methods into a class instead, so that we can more easily incorporate a retry policy (as an instance variable, because it's stateful).

You can see an example of this here: https://github.com/MetaMask/core/blob/main/packages/assets-controllers/src/token-prices-service/codefi-v2.ts

Copying the same basic retry policy should be fine at least for the status check. It would make this feature more resilient to network connectivity issues and sporadic API downtime.

I'm less sure about the POST request. Not sure how it acts when submitting the same deletion request twice, that might affect how we deal with retries.